### PR TITLE
Better GitLab detection from more API endpoints

### PR DESCRIPTION
**Pull request by prokopsimek**
## Motivation and Context

If the `/version` endpoints returned 403 because the user didn't provide an API key so the Scan fails with `InternalError: Unable to detect scanning strategy. It seems that the service is not implemented yet.` because it's not sure that the service is GitLab.

Now we call 2 more PUBLIC API endpoints specific for GitLab that can determine the service.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/detectors/ScanningStrategyDetector.ts [hunk 1]
```diff
@@ -169,6 +172,15 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
   };
 
   private determineGitLabRemoteServiceType = async (): Promise<ServiceType | undefined> => {
+    let serviceType = undefined;
+
+    try {
+      await this.gitLabService.listRepos();
+      await this.gitLabService.listGroups();
+    } catch (error) {
+      serviceType = ServiceType.gitlab;
```
Comment (prokopsimek) [inline src/detectors/ScanningStrategyDetector.ts:181]: move up

Comment (prokopsimek) [reply src/detectors/ScanningStrategyDetector.ts:181]: catch => return undefined

File: src/detectors/ScanningStrategyDetector.ts [hunk 2]
```diff
@@ -169,6 +172,15 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
...
+    }
+
     try {
       const response = await this.gitLabService.checkVersion();
```
Comment (prokopsimek) [inline src/detectors/ScanningStrategyDetector.ts:185]: add comment why 2 checks

**Comment by prokopsimek** [comment]
@adelkahomolova @ryzzaki I'll add tests. The PR is ready for a review of the logic.

File: src/detectors/ScanningStrategyDetector.spec.ts [hunk 3]
```diff
@@ -249,6 +250,46 @@ describe('ScanningStrategyDetector', () => {
...
+
+    it('remote private GitLab without protocol in the URL', async () => {
+      const repoPath = 'gitlab.com/DXHeroes/dx-scanner.git';
+      const nock = new GitLabNock('DXHeroes', 'dx-scanner', 'gitlab.com');
+      nock.listProjects();
+      nock.listGroups();
+      nock.checkVersion().reply(403);
+      nock.getRepoResponse(403);
+
+      const scanningStrategyDetector = await createScanningStrategyDetector({ uri: repoPath, auth: 'valid_token' });
```
Comment (adelkahomolova) [inline src/detectors/ScanningStrategyDetector.spec.ts:281]: As it returns 403, it's not valid token

```suggestion
      const scanningStrategyDetector = await createScanningStrategyDetector({ uri: repoPath, auth: 'invalid_token' });
```

File: src/detectors/ScanningStrategyDetector.spec.ts [hunk 4]
```diff
@@ -249,6 +250,46 @@ describe('ScanningStrategyDetector', () => {
...
+      expect(result).toEqual({
+        accessType: AccessType.private,
+        localPath: undefined,
+        remoteUrl: `https://${repoPath}`,
+        isOnline: true,
+        serviceType: ServiceType.gitlab,
+      });
+    });
+
+    it('remote private GitLab without protocol in the URL', async () => {
```
Comment (adelkahomolova) [inline src/detectors/ScanningStrategyDetector.spec.ts:273]:
```suggestion
    it('remote private GitLab without protocol in the URL with invalid credentials', async () => {
```
See comment below

File: src/detectors/ScanningStrategyDetector.spec.ts [hunk 5]
```diff
@@ -249,6 +250,46 @@ describe('ScanningStrategyDetector', () => {
       });
     });
 
+    it('remote private GitLab without protocol in the URL', async () => {
```
Comment (adelkahomolova) [inline src/detectors/ScanningStrategyDetector.spec.ts:253]:
```suggestion
    it('remote private GitLab without protocol in the URL with valid credentials', async () => {
```

File: src/detectors/ScanningStrategyDetector.ts [hunk 6]
```diff
@@ -152,6 +152,7 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
         }
       } catch (error) {
         this.d(error.message);
+        console.log(error.message);
```
Comment (adelkahomolova) [inline src/detectors/ScanningStrategyDetector.ts:155]: Is it necessary or you just forgot it here?

Comment (prokopsimek) [reply src/detectors/ScanningStrategyDetector.ts:155]:
```suggestion
```

**Comment by prokopsimek** [comment]
@adelkahomolova fixed

**Pull request merged**